lib/src/trie/proof_decode.rs

```diff
@@ -1218,11 +1220,11 @@ impl<T: AsRef<[u8]>> DecodedTrieProof<T> {
                             // If the child isn't present in the proof, then the proof is
                             // incomplete. While in some situations we could prove that the child
                             // is necessarily the next key, there is no way to know its full key.
-                            if children_present_in_proof_bitmap & (1 << key_before_nibble) == 0 {
+                            if children_present_in_proof_bitmap & (1 << child_num) == 0 {
                                 return Err(IncompleteProofError());
                             }
 
-                            for c in 0..key_before_nibble {
+                            for c in 0..child_num {
```

Comment:
Note that this could be `key_before_nibble` as well, and in fact it would be slightly faster to keep `key_before_nibble` (we know that the range between `key_before_nibble` and `child_num` is always empty), but using `child_num` makes this code way less confusing.